src/bitmap/iter.rs

```diff
@@ -298,3 +302,62 @@ impl RoaringBitmap {
         Ok(count)
     }
 }
+
+impl<'a> SkipTo for Iter<'a> {
+    type Iter = SkipToIter<'a>;
+
+    fn skip_to(self, n: u32) -> Self::Iter {
+        SkipToIter::new(&self.containers, n)
+    }
+}
```

Comment:
Note: using the full unmodified `self.containers` means this might skip _backwards_.

To do this optimally, we might need to somewhat re-implement `std::iter::Flatten` so we can access the `slice::Iter` directly (from which we can get the remaining slice), and access to the current container iterator. :/

Reply:
Really? self.containers is just a reference to RoaringBitmap.containers. It was added in this PR

I thought about changing iter::Iter to only contain a reference to containers, an index, and the iterator for the current container, and do the flatten part in ::next(), which I guess is what you're saying. This would also allow the SkipTo iterator to go backwards.
Still, are all these changes worth it to have a double ended iterator and cover the kinda weird case of skip_to to a point the iterator has already passed?

Reply:
> To do this optimally, we might need to somewhat re-implement `std::iter::Flatten` so we can access the `slice::Iter` directly (from which we can get the remaining slice), and access to the current container iterator.

I agree with what @Dr-Emann explains here. This is, to me, the best way to implement this iterator. `Flatten` is not hard to implement. We can have a special `Flatten` type with all the original `Flatten` type logic but dedicated to slices and bitmaps so that it can skip groups according to [the `split` high group](https://github.com/RoaringBitmap/roaring-rs/blob/c57c476b2fbe78ba4566973c1b1accab07b0f1fc/src/bitmap/util.rs#L3-L8) (skipping whole slices or bitmaps if the number we want to find is not in this group).

> I thought about changing iter::Iter to only contain a reference to containers, an index, and the iterator for the current container, and do the flatten part in ::next(), which I guess is what you're saying. This would also allow the SkipTo iterator to go backwards.

Let's forget about the `SkipTo` wrapper and only implement a non-lazy method like [`Iterator::advance_by` method](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.advance_by) that directly advances the iterator to the requested number. This method, `skip_to`, is only available on the `bitmap::Iter/IntoIter` types, so there is no need to think about [`std::iter::Rev` iterator adaptator](https://doc.rust-lang.org/stable/std/iter/struct.Rev.html).